backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/storage/backup/HybridBackupCommand.java

```diff
@@ -135,7 +135,16 @@ protected void executeCommand() {
     @Override
     public boolean performNextOperation(int completedChildCount) {
         restoreCommandState();
-        if (getParameters().getVmBackup().getPhase() == VmBackupPhase.STARTING) {
+        VmBackupPhase phase = getParameters().getVmBackup().getPhase();
+
+        if ((getParameters().getVmBackup().isStopped() && !phase.isBuckupCleaningUp())
```

Comment:
why is isBackupCleaningUp needed here? the next time performNextOperation is entered should be when RemoveSnapshotCommand finished

Reply:
Yeah, but then the flow will enter the if body, since the backup is still stopped. An alternative is to remove the is_stopped flag after it was handled. Would it be better?

Reply:
haven't thought about this too much, but I think if is_stopped is set, we can exit (if we are in a safe state) and handle the snapshot removal (if required) in endSuccessfully/endWithFailure, it feels wrong to mix handling of is_stopped in the normal flow

Reply:
It will always be required, because the snapshot is created even before the command returns from executeCommand().
Currently endWithFailure handles snapshot removal, but we don't want the HybridBackup command to fail if it is stopped, right?

Reply:
We can initiate removal in endSuccessfully in this case, if it makes sense not to wait for the removal of the snapshot when the user cancels 

So if we ignore transfer for a moment, if we get to performNextOperation and is_stopped is set, we can return false, and perform cleanup which is just snapshot removal

If we get to performNextOperation in the READY phase, and transfer is running it is more complicated, transfer has access to the backup and can check if it's stopped but we may need to hold the HybridBackupCommand active until TransferDiskImage is fully finalized (this can be done separately)

Reply:
@bennyz  Done, can you take a look?
As for image transfer detection, would it make sense to add another flag to the backup entity transfer_in_progress, and to monitor it the same as we monitor is_stopped, or just try to find it by backup_id in the DB?

Reply:
I don't think another flag is needed, we already fetch the backup in TransferDiskImage so we will have access to the phase

Reply:
But we don't want to change the phase externally, no? That's why this patch was initiated in the first place :)

Reply:
no, we check if there is an active backup and cancel the transfer if it's active, but let's how it behaves first